widget/form.go

```diff
@@ -134,10 +157,25 @@ func (f *Form) setUpValidation(widget fyne.CanvasObject, i int) {
 		w.SetOnValidationChanged(func(err error) {
```

Comment:
@andydotxyz comment `f.Items[i].validationError = w.Validate()` to avoid showing error in the same time user enters to the form.

Reply:
I tried that but unfortunately I cannot. The value is only updated when the error changes, and that error will never change until the validation passes and then fails again.
So if we leave it off initially then it is off past the point that it is useful

Reply:
I would love a better solution to this - I just can't think of one!

Reply:
I am not understanding

Reply:
https://user-images.githubusercontent.com/12239342/104376222-b5921280-54f2-11eb-89ae-781b62a45781.mp4

Reply:
or are you talking about that the error is shown even when the underline is not red?

Reply:
Hold on maybe you have hit something I failed to solve (the red I can live with for now - that is harder to solve)

Reply:
Ah no I remember now. If you do that then the Submit button is enabled initially when it should not be

Reply:
Hmm what if by default Submit is enabled when everything is empty. When user press Submit, Validate() on every form item is called, so Submit will be disabled at that time, and then it will work as always.

Reply:
That feels like a step backwards.

Can you share the diff for the code running above? I cannot figure out what you did to have it change on first key press

Reply:
I just commented this line `f.Items[i].validationError = w.Validate()`

Reply:
When I do that it does not update until I have entered a valid email and then backed out again

Reply:
That's weird.
<img width="485" alt="Captura de Pantalla 2021-01-12 a la(s) 16 42 06" src="https://user-images.githubusercontent.com/12239342/104377873-1d495d00-54f5-11eb-9edd-7462feca28c5.png">

Reply:
Hmm, there is magic in `w.Validate()` I did not see before

Reply:
Yes, w.Validate() launch errors in Validatable, then it should not be called at first time.

Reply:
I have a solution that works for the current button behaviour and hiding initially, just a sec

Reply:
Yeah me too
<img width="755" alt="Captura de Pantalla 2021-01-12 a la(s) 16 52 21" src="https://user-images.githubusercontent.com/12239342/104378796-8c738100-54f6-11eb-9db2-b2e59b478122.png">

Reply:
https://user-images.githubusercontent.com/12239342/104378994-ce042c00-54f6-11eb-9fbc-8bee8784d446.mp4

Reply:
I think your one could still be wrong if there were two validators and you just fixed the first one...
I am about to push a fix that should handle all cases - essentially the same trick though :)

Reply:
needs optimization as that loop logic can be done in the previous f.Items loop, but that's the idea

Reply:
> I think your one could still be wrong if there were two validators and you just fixed the first one...

I don't understand you 😅

> I am about to push a fix that should handle all cases - essentially the same trick though :)

Great! 😄

Reply:
(the problem with only calling SetValidationError(nil) and not caching the previous result was that the callback would set it to nil - with your workaround line at the end the initial state is correct, but further checks will get the wrong nil values. This meant that it was OK when you were editing the only validated field but would give a false positive if you fixed the first of many).

Anyhow - happy with this result?

Reply:
> but further checks will get the wrong nil values. 

I don't think so, maybe I am wrong. When I initially validate and set error to nil, nothing changes (and it is called just once when Renderer is created) as validation is trigger on text changed, right?

Reply:
Ignore my last comment, yes it is more flexible, the way you do 👍
I think now it looks good, I don't think it needs more changes, do it?

Reply:
I think we are good, thanks :)

Reply:
You're welcome. I'm glad to know that I can help with something. 👍